src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,842 @@
+func saveResourceUsageData(ps *stepParams, conf *config.Config) step {
+	return func() error {
+		res, err := system.GatherMetrics(ps.fs, ps.timeout, *conf)
+		if system.IsErrRedpandaDown(err) {
+			return fmt.Errorf("omitting resource usage metrics: %w", err)
+		}
+		if err != nil {
+			return fmt.Errorf("error gathering resource usage metrics: %w", err)
+		}
+		bs, err := json.Marshal(res)
+		if err != nil {
+			return fmt.Errorf("couldn't encode resource usage metrics: %w", err)
+		}
+		return writeFileToZip(ps, "resource-usage.json", bs)
+	}
+}
+
+// Queries 'pool.ntp.org' and writes a file with the reported RTT, time & precision.
+func saveNTPDrift(ps *stepParams) step {
+	return func() error {
+		host := "pool.ntp.org"
+
+		response, err := ntp.Query(host)
+		if err != nil {
+			return fmt.Errorf("error querying '%s': %w", host, err)
+		}
+
+		result := struct {
+			Host            string    `json:"host"`
+			RoundTripTimeMs int64     `json:"roundTripTimeMs"`
+			Time            time.Time `json:"time"`
+			PrecisionMs     int64     `json:"precisionMs"`
+		}{
+			Host:            host,
+			RoundTripTimeMs: response.RTT.Milliseconds(),
+			Time:            response.Time,
+			PrecisionMs:     response.Precision.Milliseconds(),
+		}
+
+		marshalled, err := json.Marshal(result)
+		if err != nil {
+			return fmt.Errorf("couldn't marshal the NTP response: %w", err)
+		}
+
+		return writeFileToZip(
+			ps,
+			"ntp.txt",
+			marshalled,
+		)
+	}
+}
+
+func saveSyslog(ps *stepParams) step {
+	return func() error {
+		entries, err := syslog.ReadAll()
+		if err != nil {
+			return err
+		}
+		return writeFileToZip(ps, "syslog.txt", entries)
+	}
+}
+
+// Queries the given admin API address for prometheus metrics.
+func savePrometheusMetrics(ps *stepParams, api *admin.AdminAPI) step {
+	return func() error {
+		raw, err := api.PrometheusMetrics()
+		if err != nil {
+			return fmt.Errorf("unable to fetch metrics from the admin API: %w", err)
+		}
+		return writeFileToZip(ps, "prometheus-metrics.txt", raw)
+	}
+}
+
+// Saves the output of `dig`
+func saveDNSData(ps *stepParams) step {
+	return func() error {
+		return writeCommandOutputToZip(ps, "dig.txt", "dig")
+	}
+}
+
+// Saves the disk usage total within redpanda's data directory.
+func saveDiskUsage(ps *stepParams, conf *config.Config) step {
+	return func() error {
+		return writeCommandOutputToZip(
+			ps,
+			"du.txt",
+			"du", "-h", conf.Redpanda.Directory,
+		)
+	}
+}
+
+// TODO: What if running inside a container/ k8s?
+// Writes the journald redpanda logs, if available, to the bundle.
+func saveLogs(ps *stepParams, since, until string, logsLimitBytes int) step {
+	return func() error {
+		args := []string{"--no-pager", "-u", "redpanda"}
+		if since != "" {
+			args = append(args, "--since", since)
+		}
+		if until != "" {
+			args = append(args, "--until", until)
+		}
+		return writeCommandOutputToZipLimit(
+			ps,
+			"redpanda.log",
+			logsLimitBytes,
+			"journalctl",
```

Comment:
Can we make a followup issue for redpanda not to dump sasl configuration on startup? (maybe it doesn't already, but it's worth triple checking, because I remember redpanda straight up dumping the redpanda.yaml file to logs)

Reply:
For sure, feel free to create it!

Reply:
https://github.com/vectorizedio/redpanda/issues/2247